hugolib/pagemeta/page_frontmatter.go

```diff
@@ -59,6 +60,16 @@ type FrontMatterDescriptor struct {
 	// May be set from the author date in Git.
 	GitAuthorDate time.Time
 
+	// This the titlefunc used by page site
+	// todo: would have preferred to point to Site to read other configuration
+	// but cannot use hugolib from nested package
+	TitleFunc func(s string) string
+
+	// Set if a title is found by handlers
+	// If have a value it will be set onto the Page title
+	Title *string
```

Comment:
Tried to find a way to map title in like done for dates via go embedded structs but doing so triggers a massive conflict on page.title vs page.Title hence I went for this minimal change approach instead.